### PR TITLE
Bounty 016: Block explorer

### PR DESCRIPTION
**Pull request by sunify**
parsec-labs/parsec-general-circle#35

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -6,7 +6,7 @@
       "transform-runtime",
       {
         "polyfill": false,
-        "regenerator": false
+        "regenerator": true
```
[the remark this answers is not in the document]
Comment (eezcjkr) [reply .babelrc]: I am actually not sure what the regenerator does, I just changed that to get the async/await to work. Does this mean I can't use async/await or something else?

File: src/components/explorer/active.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import { Alert } from 'antd';
+
+import { observer, inject } from 'mobx-react';
+
+import Block from './block';
+import Address from './address';
+import Transaction from './transaction';
+
+const Active = inject('explorer')(
```
[the remark this answers is not in the document]
Comment (eezcjkr) [reply src/components/explorer/active.js]: OK

File: src/components/explorer/active.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      BLOCK: <Block />,
+      TRANSACTION: <Transaction />,
+      ADDRESS: <Address />,
+    };
+    const error = 'No results found for your search.';
+    const init = 'Syncing state with node, this may take some time...';
+
+    return (
+      <div>
+        {!explorer.success && <Alert type="error" message={error} />}
```
[the remark this answers is not in the document]
Comment (eezcjkr) [reply src/components/explorer/active.js]: OK

File: src/components/explorer/block.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    const title = `Block ${block.number}`;
+    return (
+      <Card title={title}>
+        <h3> Hash: </h3>
+        {block.hash}
+        <h3> Timestamp: </h3>
+        {block.timestamp}
+        <h3> Transactions: </h3>
+        <TransctionList txs={block.transactions} />
+        <Button onClick={() => explorer.search(block.number - 1)}>
```
[the remark this answers is not in the document]
Comment (eezcjkr) [reply src/components/explorer/block.js]: I think ESLINT did this. Will fix.

File: src/components/explorer/transaction.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import PropTypes from 'prop-types';
+import { Card } from 'antd';
+
+import { observer, inject } from 'mobx-react';
+
+const Transaction = inject('explorer')(
+  observer(({ explorer }) => {
+    const tx = explorer.current;
+    /* eslint-disable */
+    const searchable = text => {
```
[the remark this answers is not in the document]
Comment (eezcjkr) [reply src/components/explorer/transaction.js]: OK

File: src/utils/index.js [hunk 6]
```diff
@@ -6,6 +6,12 @@ export const NETWORKS = {
   4447: { name: 'Truffle', provider: 'http://localhost:9545' },
 };
 
+export const PNODES = {
```
[the remark this answers is not in the document]
Comment (eezcjkr) [reply src/utils/index.js]: OK

File: .babelrc (hunk 1 above)
[the remark this answers is not in the document]
Comment (sunify) [reply .babelrc]: Yes, it's a kinda polyfill for `async/await` (you definitely should to what babel transpile async/await code :-))

Let's skip it for now, we can fix it at the end (or I can by myself, it's ok)

**Comment by troggy** [comment]
rebase needed

**Comment by eezcjkr** [comment]
Last commit is broken, forgot to test after rebase and now nothing works...

**Comment by eezcjkr** [comment]
Changed async/await to promises here:
https://github.com/parsec-labs/parsec-bridge-ui/commit/234437bae620129480de3d76f2bfb407d4596e16

**Pull request merged**